src/dmd/backend/gloop.d

```diff
@@ -289,54 +289,59 @@ int blockinit()
 
 void compdom()
 {
-    uint cntr;
-    vec_t t1;
-    list_t bl;
-    bool chgs;
-    block *sb;
+    compdom(dfo[0 .. dfotop]);
+}
+
+private extern (D) void compdom(block*[] dfo)
+{
+    assert(dfo.length);
+    block* sb = dfo[0];                  // starting block
 
-    assert(dfo);
-    sb = dfo[0];                         // starting block
-    t1 = vec_calloc(vec_numbits(sb.Bdom));       // allocate a temporary
     vec_clear(sb.Bdom);
     vec_setbit(0,sb.Bdom);               // starting block only doms itself
-    for (int i = 1; i < dfotop; i++)     // for all except startblock
-        vec_set(dfo[i].Bdom);            // dominate all blocks
-    cntr = 0;                            // # of times thru loop
+    foreach (b; dfo)                     // for all except startblock
+    {
+        if (b != sb)
+            vec_set(b.Bdom);             // dominate all blocks
```

Comment:
Wouldn't the following be a more literal translation that saves checking the condition on every loop iteration?

```
if (dfo.length > 0)
foreach (b; dfo[1 .. $])
  ...
```

Reply:
I was thinking about making the algorithm independent of `dfo` being an array.